source/hp/dof_handler.cc

```diff
@@ -1951,13 +1982,14 @@ namespace hp
       }
 
     Assert(levels.size() == tria->n_levels(), ExcInternalError());
-
-    // Resize active_fe_indices vectors. use zero indicator to extend
     for (unsigned int i = 0; i < levels.size(); ++i)
       {
+        // Resize active_fe_indices vectors. Use zero indicator to extend.
         levels[i]->active_fe_indices.resize(tria->n_raw_cells(i), 0);
-        levels[i]->p_refine_flags.resize(tria->n_raw_cells(i), false);
-        levels[i]->p_coarsen_flags.resize(tria->n_raw_cells(i), false);
+
+        // Resize p-flags vectors. Clear all flags after refinement finished.
+        levels[i]->p_refine_flags.assign(tria->n_raw_cells(i), false);
```

Comment:
out of curiosity, any reason to prefer `assign` to ` resize` here? if yes, why not change it on the first line?

Reply:
- `resize` changes the vector's size and initializes new entries with the provided value.
- `assign` changes the vector's size and sets all entries to the provided value.

We want to get rid of all p_refine and p_coarsen flags here, but want to keep all previous active_fe_indices.

Reply:
oh, right, it will only reset new entries to the given value. I completely missed that, thanks for clarification.